spec/multi_patient_api/bulk_data_authorization_spec.rb

```diff
@@ -0,0 +1,238 @@
+require_relative '../../lib/multi_patient_api/bulk_data_authorization.rb'
+require_relative '../../lib/multi_patient_api/bulk_data_utils.rb'
+
+RSpec.describe MultiPatientAPI::BulkDataAuthorization do
+  include BulkDataUtils
+
+  let(:group) { Inferno::Repositories::TestGroups.new.find('bulk_data_authorization') }
+  let(:session_data_repo) { Inferno::Repositories::SessionData.new }
+  let(:test_session) { repo_create(:test_session, test_group_id: 'bulk_data_authorization') }
+  let(:bulk_token_endpoint) { 'http://example.com/fhir' }
+  let(:bulk_encryption_method) { 'ES384' }
+  let(:bulk_scope) { 'system/Patient.read' }
+  let(:bulk_client_id) { 'clientID' }
+  let(:exp) { 5.minutes.from_now }
+  let(:jti) { SecureRandom.hex(32) }
+  let(:default_input) do
+    {
+      bulk_token_endpoint: bulk_token_endpoint,
+      bulk_encryption_method: bulk_encryption_method,
+      bulk_scope: bulk_scope,
+      bulk_client_id: bulk_client_id
+    }
+  end
+  let(:client_assertion_input) do
+    {
+      encryption_method: bulk_encryption_method,
+      iss: bulk_client_id,
+      sub: bulk_client_id,
+      aud: bulk_token_endpoint,
+      exp: exp,
+      jti: jti
+    }
+  end
+
+  def run(runnable, inputs = {})
+    test_run_params = { test_session_id: test_session.id }.merge(runnable.reference_hash)
+    test_run = Inferno::Repositories::TestRuns.new.create(test_run_params)
+    inputs.each do |name, value|
+      session_data_repo.save(test_session_id: test_session.id, name: name, value: value)
+    end
+    Inferno::TestRunner.new(test_session: test_session, test_run: test_run).run(runnable)
+  end
+
+  # TODO: After TLS tester class is implemented, create this test
+  describe 'endpoint TLS tests' do
+    
+  end
+
+  describe '[Invalid grant_type] test' do
+    let(:runnable) { group.tests[1] }
+    let(:client_assertion) { create_client_assertion(client_assertion_input) }
+    let(:body) do
+      {
+        'scope' => bulk_scope,
+        'grant_type' => 'not_a_grant_type',
+        'client_assertion_type' => 'urn:ietf:params:oauth:client-assertion-type:jwt-bearer',
+        'client_assertion' => client_assertion.to_s
+      }.compact
+    end
+
+    it 'passes when token endpoint requires valid grant_type' do
+      stub_request(:post, bulk_token_endpoint)
+        .with(body: body)
+        .to_return(status: 400, body: "", headers: {})
+
+      allow_any_instance_of(runnable).to receive(:create_client_assertion).and_return(client_assertion)
+      result = run(runnable, default_input)
+
+      expect(result.result).to eq('pass')
+    end
+
+    it 'fails when token endpoint allows invalid grant_type' do
+      stub_request(:post, bulk_token_endpoint)
+        .with(body: body)
+        .to_return(status: 200, body: "", headers: {})
+
+      allow_any_instance_of(runnable).to receive(:create_client_assertion).and_return(client_assertion)
+      result = run(runnable, default_input)
+      
+      expect(result.result).to eq('fail')
```

Comment:
For negative tests, you should also make an assertion on the result message to make sure that the test is failing for the correct reason.